src/requirements.cpp

```diff
@@ -125,7 +125,13 @@ std::string item_comp::to_string( const int batch, const int avail ) const
     const int c = std::abs( count ) * batch;
     const auto type_ptr = item::find_type( type );
     if( type_ptr->count_by_charges() ) {
-        if( avail > 0 ) {
+        if( avail == 2147483647/* int32 max */ ) {
```

Comment:
If you're going to use this twice, assign it to a variable, so it's not just a magic number.

Reply:
```suggestion
        if( avail == std::numeric_limits<int>::max() ) {
```
Alternatively 
(The appropriate header is included)